File: nestjs/notifications-service/src/notifications/notifications.controller.ts
import { BadRequestException, Controller, Get, Query } from '@nestjs/common';
import { NotificationsService } from './notifications.service';

@Controller('notifications')
export class NotificationsController {
  constructor(private readonly notificationsService: NotificationsService) {}

  @Get()
  async getNotifications(@Query('orderId') orderId?: string) {
    if (!orderId || orderId.trim().length === 0) {
      throw new BadRequestException('orderId query parameter is required');
    }

    return this.notificationsService.getNotifications(orderId);
  }
}

File: dotnet/src/NotificationsService/NotificationsService.Api/Controllers/NotificationsController.cs
using Microsoft.AspNetCore.Mvc;
using NotificationsService.Api.Services;

namespace NotificationsService.Api.Controllers;

[ApiController]
[Route("notifications")]
public sealed class NotificationsController(INotificationsQueryService notificationsQueryService) : ControllerBase
{
    [HttpGet]
    public async Task<IActionResult> Get([FromQuery] Guid? orderId, CancellationToken cancellationToken)
    {
        if (orderId is null || orderId == Guid.Empty)
        {
            return BadRequest(new { message = "orderId query parameter is required." });
        }

        var notifications = await notificationsQueryService.GetByOrderIdAsync(orderId.Value, cancellationToken);

        return Ok(notifications);
    }
}
